### ISSUE TITLE
Analyzer gets very confused if you use an identifier instead of an operator

### ISSUE THREAD

**Issue by Hixie** [dart-lang/sdk#26810]
I momentarily forgot that Dart uses `^` rather than `xor` for its xor operator, and wrote some code. I got confused about the error message, and eventually simplified it to the following code:

``` dart
abstract class Key {
  @override
  int get hashCode => runtimeType.hashCode xor null.hashCode;
}

void main() { }
```

This results in the following messages from the analyzer, all of which have some problem or other:
- [ ] `[lint] Annotate overridden members (test.dart, line 4, col 53)`
  This message is bogus. The only member here is in fact annotated.
- [ ] `[error] Expected to find ';' (test.dart, line 4, col 35)`
  This message is arguably correct, but at a minimum in the wrong place. I could understand it being just before the `xor`, but it actually points just after the first `.`, where a semicolon would not be valid.
- [ ] `[error] Unexpected token '.' (test.dart, line 4, col 52)`
  This message makes no sense. If you accepted an identifier (`null`), why not accept a period after it?
- [ ] `[error] Unexpected token 'null' (test.dart, line 4, col 48)`
  This message makes no sense. If you accepted `xor`, why would you not accept `null`?
- [ ] `[warning] Undefined class 'xor' (test.dart, line 4, col 44)`
  This is the real location of the only actual error in the file, but why assume `xor` is a class? A class wouldn't be valid here either. And why is it a warning? This should be a syntax error.
- [ ] `[error] Expected a class member (test.dart, line 4, col 52)`
  I've no idea what's going on with this message. Why would a class member be useful here?
- [ ] `[lint] Always override`hashCode`if overriding`==`(test.dart, line 4, col 11)`
  This lint message is backwards. It should be telling me to implement `==`, not `hashCode`. Also, it uses different quote marks than the other messages.
- [ ] `[error] The name 'hashCode' is already defined (test.dart, line 4, col 11)`
  This is the most confusing error message of the lot. Where is it previously defined? This message is the one that comes first if you sort lexically, so it's the one I spent most time on, and it didn't help me figure out what was wrong at all.
- [ ] `[error] Variables must be declared using the keywords 'const', 'final', 'var' or a type name (test.dart, line 4, col 53)`
  There are no valiables here. I don't understand why it would be asking about them.
- [ ] `[error] Expected a class member (test.dart, line 4, col 48)`
  Why is this one listed _after_ the earlier "Expected a class member"? Even if the error message is valid, the order in which it is printed makes no sense.

I would have liked to see exactly two messages:

```
[lint] Also override 'operator ==' if overriding 'hashCode' (test.dart, line 4, col 11)
[error] Expected to find ';', but found identifier 'xor' in expression. Did you mean '^'? (test.dart, line 4, col 44)
```

**Comment by pq** [comment]
cc @bwilkerson

**Comment by lrhn** [comment]
A quick guess at why some of the errors are shown: At the "expected ';' here" point, it seems the analyzer tried to actually insert a semicolon. The remaining errors are derived from that (`xor` is not a class, `null` is a keyword, so it can't be part of a signature, so it's ignored, so is the following `.`, then comes the name `hashCode` again, which is now defined twice, the second one without an `@override` annotation, etc.).

Basically, it looks like incorrect error-recovery that doesn't suppress further error reporting.

**Comment by srawlins** [comment]
This still produces quite a list of errors:

```
$ dart 26810.dart 
26810.dart:3:44: Error: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
Try adding the name of the type of the variable or the keyword 'var'.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                           ^^^
26810.dart:3:44: Error: Expected ';' after this.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                           ^^^
26810.dart:3:48: Error: Expected a class member, but got 'null'.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                               ^^^^
26810.dart:3:52: Error: Expected a class member, but got '.'.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                                   ^
26810.dart:3:53: Error: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
Try adding the name of the type of the variable or the keyword 'var'.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                                    ^^^^^^^^
26810.dart:3:53: Error: 'hashCode' is already declared in this scope.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                                    ^^^^^^^^
26810.dart:3:11: Context: Previous declaration of 'hashCode'.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
          ^^^^^^^^
26810.dart:3:35: Error: Expected ';' after this.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                  ^^^^^^^^
26810.dart:3:44: Error: Expected a class member, but got 'xor'.
  int get hashCode => runtimeType.hashCode xor null.hashCode;
                                           ^^^
```
